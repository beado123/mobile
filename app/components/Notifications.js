import React, {Component} from 'react';
import { Buffer } from 'buffer'
import { StyleSheet, Text, View, Image, Button, TouchableOpacity, FlatList, Linking, AsyncStorage } from 'react-native';
import { List, ListItem } from "react-native-elements";
import styleRepo from '/home/yihan/AwesomeProject/app/styles/Repository';

export default class NotificationsPage extends React.Component {

    state = {
        data: [],
        user: '',
        username: '',
        psw: '',
        refreshing: false,
    };

    componentWillMount(){
        this.fetchData();
    }

    /*This function sets up the data needed when loading the notifications view. */
    fetchData = async () => {
        const username = await AsyncStorage.getItem('username');
        const password = await AsyncStorage.getItem('password');
        const { params } = this.props.navigation.state;
        const login = params ? params.name : username;
        this.setState({username: username, user: login, psw: password});
        const response = await fetch('https://api.github.com/notifications?all=true' , {
          headers: {
            'Authorization': 'Basic ' + new Buffer(this.state.user + ':' + this.state.psw).toString('base64'),
            'Content-Lenght': 0,
          }
      });
        let responseJson = await response.json();
        this.setState({data: responseJson});
    }

    /*This function handles refreshing of current list. */
    handleRefresh = () => {
        this.setState({refreshing: true},() => {
            this.fetchData();
        });
    };

    render() {
        return (
            <View style={styleRepo.container}>
                <Text style={styleRepo.header}>notifications</Text>
                <List>
                    <FlatList
                        data={this.state.data}
                        extraData={this.state}
                        refreshing = {this.state.refreshing}
                        onRefresh = {this.handleRefresh}
                        keyExtractor={(item, index) => index}
                        renderItem={({ item }) => (
                            <ListItem
                                hideChevron = {true}
                                subtitle={item.subject.title }
                                subtitleNumberOfLines = {2}
                                titleStyle={{fontSize:17}}
                                rightTitle = {item.updated_at.slice(0,-1).split('T')[0]+' '+item.updated_at.slice(0,-1).split('T')[1]}
                                rightTitleNumberOfLines = {2}
                                rightTitleStyle = {{width: 80}}
                                rightTitleContainerStyle = {{width:80}}
                                title = {item.repository.owner.login+'/'+item.repository.name}
                                titleNumberOfLines = {2}
                                titleContainerStyle	= {{width:200}}

                            />
                        )}
                    />
                </List>
                <View style={styleRepo.footer}>
                    <TouchableOpacity style={styleRepo.gobackButton}>
                        <Text style={{color:'#fff'}} onPress={() => this.props.navigation.goBack()}> Go Back </Text>
                        </TouchableOpacity>
                </View>
            </View>
        );
    }
}
