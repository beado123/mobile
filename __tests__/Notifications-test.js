import React from 'react';
import Notifications from '/home/yihan/AwesomeProject/app/components/Notifications.js';
import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const tree = renderer.create(<Notifications />).toJSON();
  expect(tree).toMatchSnapshot();
});
