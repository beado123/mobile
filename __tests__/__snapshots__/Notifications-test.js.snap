// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "paddingTop": 10,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 20,
        "justifyContent": "center",
        "textAlign": "center",
      }
    }
  >
    notifications
  </Text>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#ffffff",
          "borderColor": "#bbb",
          "borderTopWidth": 1,
          "marginTop": 20,
        },
        undefined,
      ]
    }
  >
    <RCTScrollView
      data={Array []}
      disableVirtualization={false}
      extraData={
        Object {
          "data": Array [],
          "psw": "",
          "refreshing": false,
          "user": "",
          "username": "",
        }
      }
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={1}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onRefresh={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      refreshControl={
        <RefreshControlMock
          onRefresh={[Function]}
          progressViewOffset={undefined}
          refreshing={false}
        />
      }
      refreshing={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <RCTRefreshControl />
      <View />
    </RCTScrollView>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "bottom": 0,
        "height": 60,
        "left": 0,
        "position": "absolute",
        "right": 0,
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#64678f",
          "borderColor": "#000",
          "borderRadius": 40,
          "bottom": 0,
          "height": 60,
          "justifyContent": "center",
          "opacity": 1,
          "width": 60,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        onPress={[Function]}
        style={
          Object {
            "color": "#fff",
          }
        }
      >
         Go Back 
      </Text>
    </View>
  </View>
</View>
`;
